### PR TITLE
configuration file with all dependencies

### PR DESCRIPTION
**Pull request by albi3ro**
In order to test the devices and run all benchmarks, we need more dependencies, but we don't want to install EVERYTHING for a simple benchmark.  

So if you need everything for the benchmark run, you can use
```
asv run --config full_dependencies.conf.json
```

and that will install
* pennylane-lightning
* pennylane-qiskit
* pennylane-cirq
* pennylane-qulacs
* networkx
* torch
* tensorflow

and all their dependencies.

### CODE REVIEW

File: full_dependencies.conf.json [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    // The project's homepage
+    "project_url": "pennylane.ai",
+
+    // The URL or local path of the source code repository for the
+    // project being benchmarked
+    "repo": "https://github.com/PennyLaneAI/pennylane.git",
+
+    // The directory (relative to the current directory) that benchmarks are
+    // stored in.  If not provided, defaults to "benchmarks"
+    "benchmark_dir": "benchmarks",
```
Comment (trbromley) [inline full_dependencies.conf.json:20]: Do we need to change this? Looks like the default is this (https://asv.readthedocs.io/en/stable/asv.conf.json.html#benchmark-dir)

File: full_dependencies.conf.json [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    // followed by the pip installed packages).
+    //
+    "matrix": {
+        "tensorflow": [],
+        "torch": [],
+        "networkx": [],
+        "git+https://github.com/PennyLaneAI/pennylane-lightning.git": [],
+        "git+https://github.com/PennyLaneAI/pennylane-qiskit.git": [],
+        "git+https://github.com/PennyLaneAI/pennylane-cirq.git": [],
+        "git+https://github.com/PennyLaneAI/pennylane-qulacs.git": [],
```
Comment (trbromley) [inline full_dependencies.conf.json:58]: Ah was this linked to the question of comparing `master-master` vs `master-release`? I also agree `master-master` probably makes the most sense.

File: full_dependencies.conf.json [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+    // The Python project's subdirectory in your repo.  If missing or
+    // the empty string, the project is assumed to be located at the root
+    // of the repository.
+    //"repo_subdir": "benchmarks",
+
+    // Customizable commands for building, installing, and
+    // uninstalling the project. See asv.conf.json documentation.
+    //
+    // "install_command": ["in-dir={env_dir} python -mpip install {wheel_file}"],
+    // "uninstall_command": ["return-code=any python -mpip uninstall -y {project}"],
```
Comment (trbromley) [inline full_dependencies.conf.json:73]:
```suggestion
    // "install_command": ["in-dir={env_dir} python -m pip install {wheel_file}"],
    // "uninstall_command": ["return-code=any python -m pip uninstall -y {project}"],
```
I know these aren't used, but I was curious if they'd work without the space.

**Review by trbromley — APPROVED** [review]
Thanks @albi3ro! Looks good, although I'm still not too familiar with the benchmark suite.

**Comment by albi3ro** [comment]
Thanks @trbromley  for the review, but I've found a different method for managing the dependencies by creating a custom environment.

**Pull request closed without merging**